features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/messagecomposer/MessageComposerPresenter.kt

```diff
@@ -389,6 +391,28 @@ class MessageComposerPresenter @Inject constructor(
                         }
                     }
                 }
+                is MessageComposerEvents.InsertMention -> {
+                    localCoroutineScope.launch {
+                        room.membersStateFlow.collect {
+                            val member = it.joinedRoomMembers().find { it.userId == event.userId } ?: return@collect
+                            if (showTextFormatting) {
+                                val text = member.userId.value
+                                val link = permalinkBuilder.permalinkForUser(member.userId).getOrNull() ?: return@collect
+                                // FIXME: This should use the un-exported `insertMention()`, probably. Currently it fails because there's no active `suggestion`
+                                richTextEditorState.insertMentionAtSuggestion(text = text, link = link)
+                            } else {
+                                val end = markdownTextEditorState.selection.last
+                                // Create a suggestion at the current selection (or end) so insertSuggestion() knows where to add the pill
+                                markdownTextEditorState.currentSuggestion = Suggestion(end, end, SuggestionType.Mention, "")
```

Comment:
This feels like a bit of a hack; not sure I like it. Maybe it would be better to move this to `MarkdownTextEditorState` and tidy it up a bit